content/news/XXXX-XX-XX-gsoc-final-blog-fatih-emre-yildiz.md

```diff
@@ -0,0 +1,395 @@
+![Find On Web Feature]({static}/images/news/findonwebfeature.png)
+
+This Implementation was designed roughly for the next step. The next step was:
+
+##### *Find On Web With Factory Design Pattern*
+This implementation aims to have a Factory Design Pattern for the Find On Web feature. 
+
+As its name states, you can think about you have a factory and you can produce products. Our factory is the Find On Web itself and our products are the services. The most important point here is that all products coming out of a factory must be derived from the same interface or class. Thanks to this pattern, you can abstract our Services for future use. you can produce products - add new services - straight forward.
+
+Since there are many other online music databases, that is nice to have Find On Web feature with the Factory Design Pattern. Just in case if needed any other service can be added easily without making any changes in the code base.
+
+Related Files:
+
+* [findonwebfactory.cpp](https://github.com/mixxxdj/mixxx/blob/main/src/widget/findonwebmenufactory.cpp)
+* [findonwebfactory.h](https://github.com/mixxxdj/mixxx/blob/main/src/widget/findonwebmenufactory.h)
+* [Find On Web Menu Services Folder](https://github.com/mixxxdj/mixxx/tree/main/src/widget/findonwebmenuservices)
+
+##### *Editable Track Menu*
+While discussing the `Find On Web` feature on Zulip. There was an idea which was the Track Menu getting crowded and we might consider making it editable. At first I thought the same and this feature would be nice to have and that can help me to get to know the code base for the preference option. So I tried to make it, just to learn how to deal with the preferences. After I introduced this idea on the Zulip channel, that was rejected by the community. So we have decided to not implement this.
+
+##### *Cover Art Fetcher*
+This is the third side project for the Track Suggestion Feature. Fetching cover art from online resources. According to the search we have done, we decided to fetch the cover art from Cover Art Archive (CAA) hosted by https://archive.org. The Cover Art Archive is a joint project between the Internet Archive and MusicBrainz.
+
+The CAA uses the track's Release MBID (MusicBrainz Identifier) to retrieve cover art. For that, the `Import Metadata From Musicbrainz` feature could be used.
+
+While Implementing this, At first, I sent the request as described in the API documentation. I was having the response but not correctly, it was saying that the redirected response is received. At first, I tried to solve this issue by sending another request to the redirected URL which was not possible. Then I stopped for a moment and thought about what I can do in this situation, then I found out that QT's class QNetworkAccessManager has [Redirect Policy](https://doc.qt.io/qt-6/qnetworkrequest.html#RedirectPolicy-enum) already. After a long time trying to fix this issue, that was surprising that the solution was in front of me and that was simple.
```

Comment:
.. that the solution was in front of me and that was simple.

I think the sentence needs some love. What expect the technical reader what expects a Mixxx user to read?

Reply:
Okay, I'll take a look.